pandas/core/frame.py

```diff
@@ -9883,6 +9885,9 @@ def join(
         sort : bool, default False
             Order result DataFrame lexicographically by the join key. If False,
             the order of the join key depends on the join type (how keyword).
+
+            .. note::
```

Comment:
Not sure if mentioning this is a good idea, I would keep it out of our docs. Otherwise we would have to basically include this everywhere